### ISSUE TITLE
SDK 29 ExpoKit Android app does not load bundled assets

### ISSUE THREAD

**Issue by llamaluvr** [expo/expo#2105]
This is an off-shoot of https://github.com/expo/expo/issues/2060 / https://forums.expo.io/t/bundling-assets-for-offline-not-working-on-expokit-29/12576/2. Originally, my issue was that bundled assets were not working offline for iOS or Android on SDK 29 / ExpoKit. Working with @sjchmiela, we were able to get bundling working again on iOS by adding the `import { Assets } from 'expo'; Assets;` code to App.js and by updating the Podfile to version 2.7.4. 

However, the equivalent fix for Android (same code, ensure expokit NPM package is 1.4.0) is not working. Bundled assets do not load at all (even when online - with the exception of a few small ones intermittently). Images referenced via an internet URL work just fine.

Before I added the following code,

```
import {Assets} from 'expo';
Assets;
```

to App.js, the same thing happened whether I loaded the app offline or online: the image would not appear. 

Since adding that code, if I start the app offline, I get the Expo blue screen and the error: "Uncaught Error: Could not load request from exp://exp.host/@username/slug: the request timed out". If I load the app online, I just get the blank screen without the image.

### Environment

ExpoKit / Android / SDK 29

Diagnostics:
```
Environment:
  OS: macOS High Sierra 10.13.6
  Node: 8.11.2
  Yarn: 1.6.0
  npm: 5.6.0
  Watchman: 4.9.0
  Xcode: Xcode 9.4.1 Build version 9F2000
  Android Studio: 3.0 AI-171.4443003

Packages: (wanted => installed)
  expo: ^29.0.0 => 29.0.0
  react: 16.3.1 => 16.3.1
  react-native: https://github.com/expo/react-native/archive/sdk-29.0.0.tar.gz => 0.55.4

Diagnostics report:
  https://exp-xde-diagnostics.s3.amazonaws.com/nudge-2a70a585-cdd8-4b6f-a737-ff2ecb3560ea.tar.gz
```

### Steps to Reproduce

(Write your steps here:)

1. Create new app with `expo init` (blank template)
2. Add an image to the assets folder. Display that image in App.js.
3. Run expo start.
4. Run the app on a device in Android Studio (debug mode).
5. You’ll see a gray screen with a beer mug icon in the middle of it.
7. Run expo publish.
8. In Android Studio, go to Build -> Generate Signed APK, follow the prompts to build an APK.
9. Install the APK (I’m using the “prodMinSDK prodRelease” APK… I think that’s the one I want to use?).
10. Start up the app.

### Expected Behavior

I should see the beer mug image, even if the phone is in airplane mode.

(Write what you thought would happen.)

### Actual Behavior

Prior to adding `import { Asset } from 'expo'; Asset;` code to App.js:

- When loading the app with the phone offline or online, no beer mug icon, just a blank screen.

After adding the code to App.js:
- Got the "request timed out error" noted above when loading offline
- No image, just gray screen when loading online.

### Reproducible Demo

I’ve made a minimal example demonstrating this issue here: 
https://github.com/llamaluvr/test-asset-bundling.  This example is based on the “blank” starting point when you run expo init. All I did was add a single png image asset and change App.js to show that image in them middle of the screen. The example already has an assetBundlePattern and an assets folder in app.json that I did not change.

**Comment by audiolion** [comment]
+1 I just tried to build my android release today and ran into this, I was so confused and thought I must have done something wrong, but this happens in any detached project from sdk 29!

**Comment by sjchmiela** [comment]
Hi guys! I've just submitted an upstream pull request fixing this bug. Sorry for the trouble!

Until it's merged and released I encourage you to replace your current `expo.gradle` with https://gist.github.com/sjchmiela/c73b16008d886f2f55308c7e495479f7.

The bug started to occur when we upgraded Gradle plugin to version 3. We then introduced product flavor dimensions which this script didn't take into account when calculating possible build variants. It turns out we can just use `android.applicationVariants` to grab all the possible build variants and attach the bundling code to them.
